tests/backend/test_config.py

```
@@ -0,0 +1,106 @@
+    assert not new_path.exists()
+
+
+def test_rename_unlink_failure_leaves_two_complete_sessions(
+    config_dir: Path, monkeypatch: pytest.MonkeyPatch
+) -> None:
+    """Propagate old-file cleanup failure after writing the complete new file."""
+    config.save_session({"label": "Old", "value": 1})
+    old_path = config.get_session_path("Old")
+    real_unlink = Path.unlink
+
+    def fail_old_unlink(path: Path, *args: Any, **kwargs: Any) -> None:
```

Ruff (0.15.21)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `*args`

(ANN401)